extends Node


#Handle User Created Level Page

var apiUrl_UCL = "ucl/all/all"
var result_UCL
var responseCode_UCL
var base_index = 0
var max_index
var nextPage = "true"
var number_to_minus

func _ready():
	root.set_screen_orientation(0)
	render_ucl()
	
	$newLevel.connect("pressed", self, "newLevel_pressed")
	$nextPage.connect("pressed", self, "nextPage_pressed")
	$previousPage.connect("pressed", self, "previousPage_pressed")
	
	for button in $UCLs.get_children():
		button.connect("pressed", self, "handleSelectedUCL", [button])
	pass 

func newLevel_pressed():
	root.switch_scene("res://entities/UCL/CreateNewLevel/CreateNewLevelView.tscn")

func _notification(what):
    if (what == MainLoop.NOTIFICATION_WM_GO_BACK_REQUEST):
        root.return_to_last()
		
func render_ucl():
	$loading.popup()	
	for UCLTitle in $UCLTitle.get_children():
		UCLTitle.clear()

	apiController.apiCallGet(apiUrl_UCL)
	yield(apiController, "request_completed")
	
	result_UCL = apiController.getResult()
	responseCode_UCL = apiController.getResponseCode()
	
	max_index = result_UCL.size()
	number_to_minus = 6

	
	for i in range(1, 7):
		if (i == 1):
			
			if (base_index >= max_index):
				nextPage = "false"
			else:
				$UCLTitle/UCLTitle1.append_bbcode(result_UCL[base_index]["uclName"])
				print(result_UCL[base_index]["uclName"])
				base_index = base_index + 5
				number_to_minus = number_to_minus + 5
				$loading.hide()
			
		if (i == 2):
			
			if (base_index >= max_index):
				nextPage = "false"
			else:
				$UCLTitle/UCLTitle2.append_bbcode(result_UCL[base_index]["uclName"])
				print(result_UCL[base_index]["uclName"])
				base_index = base_index + 5
				number_to_minus = number_to_minus + 5
			
		if (i == 3):
			
			if (base_index >= max_index):
				nextPage = "false"
			else:
				$UCLTitle/UCLTitle3.append_bbcode(result_UCL[base_index]["uclName"])
				print(result_UCL[base_index]["uclName"])
				base_index = base_index + 5
				number_to_minus = number_to_minus + 5
			
		if (i == 4):
			
			if (base_index >= max_index):
				nextPage = "false"
			else:
				$UCLTitle/UCLTitle4.append_bbcode(result_UCL[base_index]["uclName"])
				print(result_UCL[base_index]["uclName"])
				base_index = base_index + 5
				number_to_minus = number_to_minus + 5
			
		if (i == 5):
			if (base_index >= max_index):
				nextPage = "false"
			else:
				$UCLTitle/UCLTitle5.append_bbcode(result_UCL[base_index]["uclName"])
				print(result_UCL[base_index]["uclName"])
				base_index = base_index + 5
				number_to_minus = number_to_minus + 5
			
		if (i == 6):
			if (base_index >= max_index):
				nextPage = "false"
			else:
				$UCLTitle/UCLTitle6.append_bbcode(result_UCL[base_index]["uclName"])
				print(result_UCL[base_index]["uclName"])
				base_index = base_index + 5
				number_to_minus = number_to_minus + 5
				nextPage = "true"
	
	if (base_index >= max_index):
		nextPage = "false"

	if (base_index <= 30):
		$previousPage.hide()
	else:
		$previousPage.show()
		
	if (nextPage == "true"):
		$nextPage.show()
	else:
		$nextPage.hide()
func previousPage_pressed():
	base_index = base_index - number_to_minus
	render_ucl()
	pass

func nextPage_pressed():
	render_ucl()
	pass